src/adapters/refract/transformSections.coffee

```diff
@@ -2,37 +2,65 @@ _ = require('./helper')
+
 transformResources = require('./transformResources')
+transformResource = require('./transformResource')
+
 
-module.exports = (element) ->
+module.exports = (parentElement) ->
   resourceGroups = []
-  resourceGroupsElements = _.chain(element)
-                    .content()
-                    .filter({element: 'category', meta: {classes: ['resourceGroup']}})
-                    .value()
-
-  # If we can't find any resource group we can assume that there are Resources directly
-  if resourceGroupsElements.length is 0
-    resourceGroups.push(
-      new blueprintApi.Section({
-        name: ''
+
+  # List of Application AST Resources (= already transformed
+  # Refract Resources into the Application AST Resource).
+  applicationAstResources = []
+
+  _.forEach(parentElement.content, (element, index) ->
+    # If the element is a resource, transform it
+    # into the Application AST Resource and push it
+    # to a temporary array, we'll create an artificial
+    # resource group (section) for these resources
+    # later in the process.
+    if element.element is 'resource'
+      applicationAstResources = applicationAstResources.concat(
+        transformResource(element)
+      )
+
+    # Element is category (resource group most probably).
+    if element.element is 'category'
+      # First let's create an artificial resource group (section)
+      # for previous resources.
+      if applicationAstResources.length
+        resourceGroups.push(new blueprintApi.Section({
+          name: ''
+          resources: applicationAstResources
+        }))
+
+        applicationAstResources = []
+
+      description = getDescription(element)
+
+      # Then create a new section in the Application AST
+      # corresponding to the Category element.
+      resourceGroup = new blueprintApi.Section({
+        name: _.chain(element).get('meta.title', '').contentOrValue().value()
+        description: description.raw
+        htmlDescription: description.html
         resources: transformResources(element)
       })
-    )
-  else
-    resourceGroupsElements.forEach((resourceGroupElement) ->
-      section = new blueprintApi.Section({
-        name: _.chain(resourceGroupElement).get('meta.title', '').contentOrValue().value()
-      })
-
-      description = getDescription(resourceGroupElement)
 
-      section.description = description.raw
-      section.htmlDescription = description.html
+      resourceGroups.push(resourceGroup)
 
-      section.resources = transformResources(resourceGroupElement)
+    # If this the last run, make sure tu flush the resources
+    # into an an artificial resource group (section).
+    isLastRun = index is (parentElement.content.length - 1)
 
-      resourceGroups.push(section)
-    )
+    if isLastRun and applicationAstResources.length
+      resourceGroups.push(
+        new blueprintApi.Section({
+          name: ''
+          resources: applicationAstResources
+        })
+      )
```

Comment:
Should this be outside of the `forEach`? Seems like you could wait until the iterating is finished and then push it to resource groups. This would allow you to not do this `isLastRun` check.